backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Orders/CreateOrderCH.cs

```diff
@@ -54,29 +72,30 @@ public CreateOrderQH(CoreDbContext dbContext)
 
         public async Task ExecuteAsync(CoreContext context, CreateOrder command)
         {
-            var result = await dbContext.Orders.AddAsync(
-                new Order(command.NewOrder.Address)
+            var address = command.Address ?? dbContext.Users.Where(u => u.Id == context.UserId).First().Address;
+            var newOrder = new Order(address)
+            {
+                UserId = context.UserId,
+                Price = context.GetShoppingCartPrice(dbContext),
+                OrderedDate = DateTime.Now,
+                OrderState = OrderState.Pending,
+            };
+            var shp = await dbContext.ShoppingCarts.Where(s => s.UserId == context.UserId).FirstOrDefaultAsync();
+            foreach (var prod in shp!.ShoppingCartProducts)
+            {
+                newOrder.OrdersProducts.Add(new OrderProduct()
                 {
-                    UserId = context.UserId,
-                    Price = command.NewOrder.Price,
-                    OrderedDate = DateTime.Now,
-                    OrderState = OrderState.Pending,
+                    Amount = prod.Amount,
+                    ProductId = prod.ProductId,
+                    OrderId = newOrder.Id
                 });
-            if (result.Entity != null)
+            }
+            var result = await dbContext.Orders.AddAsync(newOrder);
+            if(result != null)
```

Comment:
I already created a layout for taking a specific amount of specific products in the shopping cart, maybe revert this somehow?

Reply:
ok zrewertuje